This patch reverts two upstream commits due to build and/or runtime failures
introduced by them.

Commit ac042f3f58d33ce9cd5ff61750f06da1a1d7b0eb causes 'autoreconf -fi' to
fail, erroneously claiming that autoconf-archive is not installed, when in fact
it IS installed.

Commit 1a5417f39a0ccefcdd5440f2a67f84d2d2e26960 causes grub-install to fail on
Legacy BIOS (i386-pc) systems, reporting that kernel.img is miscompiled.

Both of these bad commits unfortunately made it into the GRUB 2.14 "stable"
release, despite only being published AFTER the 2.14-rc1 prerelease (thus, not
being present in said prerelease).

diff '--color=auto' -apruN a/acinclude.m4 b/acinclude.m4
--- a/acinclude.m4	2025-11-20 16:02:39.000000000 +0000
+++ b/acinclude.m4	2026-01-19 23:45:14.135084158 +0000
@@ -79,11 +79,6 @@ AC_DEFUN([grub_PROG_OBJCOPY_ABSOLUTE],
 [AC_MSG_CHECKING([whether ${TARGET_OBJCOPY} works for absolute addresses])
 AC_CACHE_VAL(grub_cv_prog_objcopy_absolute,
 [cat > conftest.c <<\EOF
-asm (
-    ".globl start, _start, __start\n"
-    ".ifdef cmain; .set start = _start = __start = cmain\n.endif\n"
-    ".ifdef _cmain; .set start = _start = __start = _cmain\n.endif\n"
-);
 void cmain (void);
 void
 cmain (void)
diff '--color=auto' -apruN a/configure.ac b/configure.ac
--- a/configure.ac	2026-01-14 14:30:04.000000000 +0000
+++ b/configure.ac	2026-01-19 23:45:14.135230510 +0000
@@ -1461,6 +1461,7 @@ elif test x$grub_cv_target_cc_link_forma
   TARGET_IMG_LDSCRIPT='$(top_srcdir)'"/conf/i386-cygwin-img-ld.sc"
   TARGET_IMG_LDFLAGS="-Wl,-T${TARGET_IMG_LDSCRIPT}"
   TARGET_IMG_LDFLAGS_AC="-Wl,-T${srcdir}/conf/i386-cygwin-img-ld.sc"
+  TARGET_IMG_BASE_LDOPT="-Wl,-Ttext"
   TARGET_IMG_CFLAGS=
 else
   TARGET_APPLE_LINKER=0
@@ -1468,6 +1469,7 @@ else
   TARGET_IMG_LDSCRIPT=
   TARGET_IMG_LDFLAGS='-Wl,-N'
   TARGET_IMG_LDFLAGS_AC='-Wl,-N'
+  TARGET_IMG_BASE_LDOPT="-Wl,-Ttext"
   TARGET_IMG_CFLAGS=
 fi
 
@@ -1792,24 +1794,7 @@ LIBS=""
 # Defined in acinclude.m4.
 grub_ASM_USCORE
 grub_PROG_TARGET_CC
-
-# The error message produced by autoconf if autoconf-archive is not installed is
-# quite misleading and not very helpful. So, try point people in the right direction.
-m4_ifndef([AX_CHECK_LINK_FLAG], [m4_fatal([autoconf-archive is missing. You must install it to generate the configure script.])])
-
 if test "x$TARGET_APPLE_LINKER" != x1 ; then
-AX_CHECK_LINK_FLAG([-Wl,--image-base,0x400000],
-    [TARGET_IMG_BASE_LDOPT="-Wl,--image-base"],
-    [TARGET_IMG_BASE_LDOPT="-Wl,-Ttext"],
-    [],
-    [AC_LANG_SOURCE([
-asm (".globl start; start:");
-asm (".globl _start; _start:");
-asm (".globl __start; __start:");
-void __main (void);
-void __main (void) {}
-int main (void);
-    ])])
 grub_PROG_OBJCOPY_ABSOLUTE
 fi
 grub_PROG_LD_BUILD_ID_NONE
